src/tests/Common/XUnitWrapperGenerator/ITestInfo.cs

```diff
@@ -0,0 +1,139 @@
+
+namespace XUnitWrapperGenerator;
+
+interface ITestInfo
+{
+    string ExecutionStatement { get; }
+}
+
+sealed class BasicTestMethod : ITestInfo
+{
+    public BasicTestMethod(IMethodSymbol method, string externAlias, ImmutableArray<string> arguments = default)
+    {
+        var args = arguments.IsDefaultOrEmpty ? "" : string.Join(", ", arguments);
+        string containingType = method.ContainingType.ToDisplayString(XUnitWrapperGenerator.FullyQualifiedWithoutGlobalNamespace);
+        if (method.IsStatic)
+        {
+            ExecutionStatement = $"{externAlias}::{containingType}.{method.Name}({args});";
+        }
+        else
+        {
+            ExecutionStatement = $"using ({externAlias}::{containingType} obj = new()) obj.{method.Name}({args});";
+        }
+    }
+
+    public string ExecutionStatement { get; }
+
+    public override bool Equals(object obj)
+    {
+        return obj is BasicTestMethod other && ExecutionStatement == other.ExecutionStatement;
+    }
+}
+
+sealed class ConditionalTest : ITestInfo
+{
+    public ConditionalTest(ITestInfo innerTest, string condition)
+    {
+        ExecutionStatement = $"if ({condition}) {{ {innerTest.ExecutionStatement} }}";
+    }
+
+    public string ExecutionStatement { get; }
+
+    public override bool Equals(object obj)
+    {
+        return obj is ConditionalTest other && ExecutionStatement == other.ExecutionStatement;
+    }
+}
+
+sealed class PlatformSpecificTest : ITestInfo
+{
+    public PlatformSpecificTest(ITestInfo innerTest, Xunit.TestPlatforms platform)
+    {
+        List<string> platformCheckConditions = new();
+        if (platform.HasFlag(Xunit.TestPlatforms.Windows))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsWindows()");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.Linux))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsLinux()");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.OSX))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsMacOS()");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.illumos))
+        {
+            platformCheckConditions.Add(@"global::System.OperatingSystem.IsOSPlatform(""illumos"")");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.Solaris))
+        {
+            platformCheckConditions.Add(@"global::System.OperatingSystem.IsOSPlatform(""Solaris"")");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.Android))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsAndroid()");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.iOS))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsIOS()");
+        }
+        if (platform.HasFlag(Xunit.TestPlatforms.tvOS))
+        {
+            platformCheckConditions.Add("global::System.OperatingSystem.IsAndroid()");
```

Comment:
copy/pase mistake, this should be 
```suggestion
            platformCheckConditions.Add("global::System.OperatingSystem. IsTVOS()");
```